TypeCobol/Compiler/Symbols/ProgramSymbol.cs

```diff
@@ -0,0 +1,473 @@
+                    value.Add(varSym);
+                }
+                //Remember a variable that maybe expanded.
+                if (varSym.HasFlag(Flags.HasATypedefType))
+                {
+                    VariableTypeSymbols.AddLast(varSym);
+                }
+            }
+            return varSym;
+        }
+
+        /// <summary>
+        /// Add the element to this Program's domain.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public VariableSymbol Add(VariableSymbol element)
+        {
+            return AddToDomain(element);
+        }
+
+        /// <summary>
+        /// Get the Scope of symbol associated to the given symbol name.
+        /// </summary>
+        /// <param name="path">The Symbol's path to get the Scope, the path is in reverse order à la COBOL.</param>
+        /// <returns>The Multi Symbol set of all symbol corresponding to the given path.</returns>
+        public Scope<VariableSymbol>.MultiSymbols Get(string[] path)
+        {
+            return ResolveReference(path, true);
+        }
+
+        /// <summary>
+        /// Is this program nested.
+        /// </summary>
+        public virtual bool IsNested => Owner != null && Kind == Kinds.Program && Owner.Kind == Kinds.Program;
+
+        /// <summary>
+        /// Get the Variable visibility mask.
+        /// </summary>
+        public virtual Flags VariableVisibilityMask => IsNested ? (Flags.Global | Flags.GLOBAL_STORAGE) : 0;
+
+        /// <summary>
+        /// Get the type visibility mask for a Program.
+        /// </summary>
+        public virtual Flags TypeVisibilityMask => IsNested ? (Flags.Global | Flags.Private | Flags.Public) : 0;
+
+        /// <summary>
+        /// Determines if a Type is accessible from this Program.
+        /// </summary>
+        /// <param name="typeSym">The Type To check</param>
+        /// <returns>true if the type is accessible, false otherwise</returns>
+        public virtual bool IsTypeAccessible(TypedefSymbol typeSym)
+        {
+            System.Diagnostics.Debug.Assert(typeSym != null);
+            Symbol typeSymPrg = typeSym.TopParent(Kinds.Program);
+            System.Diagnostics.Debug.Assert(typeSymPrg != null);
+            Symbol myTopPrg = TopParent(Kinds.Program);
+            System.Diagnostics.Debug.Assert(myTopPrg != null);
+            Flags mask = TypeVisibilityMask;
+
+            if (typeSymPrg == myTopPrg)
+            {//Same program ==> Apply the visibility mask
+                return mask == 0 || typeSym.HasFlag(mask);
+            }
+            else
+            {//Different programs ==> only public.
+                return typeSym.HasFlag(Flags.Public);
+            }
+        }
+
+        /// <summary>
+        /// Resolve the given Symbol paths from this scope
+        /// </summary>
+        /// <param name="paths">The qualified path of the symbol reference in COBOL85 order: MUST BE IN LOWER CASES</param>
```

Comment:
You can use Equals(xxx, StringComparison.OrdinalIgnoreCase) and Dictionary with StringComparer.OrdinalIgnoreCase
This will make the API easier to use and it's more performant.

Reply:
Ok right